### PR TITLE
Update macspice from 3.1.24 to 3.1.25 and complete missing fields

### PR DESCRIPTION
**Pull request by alxrey**
Created with `brew bump-cask-pr`.

### CODE REVIEW

File: Casks/macspice.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
 cask "macspice" do
-  version "3.1.24,341"
-  sha256 "486baee366773f3589c870d9294e1028c118e29f9bf4dc824322e37f2782475f"
+  version "3.1.25,343"
+  sha256 "22f82c32b6e79916b123a7d1f5ded71a9a30cffc5ab63262b43a0d11db3c45f9"
 
   url "https://www.macspice.com/mirror/binaries/v#{version.before_comma}/MacSpice3f5.dmg"
```
Comment (neutric) [inline Casks/macspice.rb:5]: `url` scheme changed => `url "https://www.macspice.com/binaries/v#{version.csv.first}/MacSpice3f5.dmg"`

**Comment by miccal** [comment]
Not sure what to suggest here -- there seems to be an issue with all the re-directs:
```
|-> curl -sLI https://www.macspice.com/mirror/binaries/v3.1.25/MacSpice3f5.dmg | grep -i Location
Location: https://www.macspice.com/binaries/v3.1.25/MacSpice3f5.dmg
Location: https://www.dropbox.com/s/op2w15yfownolim/MacSpice3f5.dmg?dl=1
location: /s/dl/op2w15yfownolim/MacSpice3f5.dmg
location: https://uc6deaae878b0d267864bc6973e0.dl.dropboxusercontent.com/cd/0/get/Bcu3BGV5TlhXqIJMPDy7QFDTm_HHjwCVo0xbLLfmDXJmoo5Gq9XBb3M62oDtvia23SsOGZ_7W9KuzuHb8rDTiH6rBAGPc6Za0hSxepL-duYo9QnCqLmxy-Y0Oi5z3ggmJHLi6H6h4ny7iBRkcLNKarwd/file?dl=1#
```

**Comment by neutric** [comment]
It downloads fine locally, both with `wget` and with `curl`. Might the 403 error code (=forbidden) be avoided with `user_agent: :fake`?

**Comment by miccal** [comment]
Weird ...

**Comment by neutric** [comment]
This installs just fine locally with

```
version "3.1.25,343"
sha256 "22f82c32b6e79916b123a7d1f5ded71a9a30cffc5ab63262b43a0d11db3c45f9"
url "https://www.macspice.com/binaries/v#{version.csv.first}/MacSpice3f5.dmg"
```

**Comment by miccal** [comment]
> This installs just fine locally with
> 
> ```
> version "3.1.25,343"
> sha256 "22f82c32b6e79916b123a7d1f5ded71a9a30cffc5ab63262b43a0d11db3c45f9"
> url "https://www.macspice.com/binaries/v#{version.csv.first}/MacSpice3f5.dmg"
> ```

It does for me too -- are you using a system `curl` or a `brew`'ed `curl`?

**Comment by neutric** [comment]
> It does for me too -- are you using a system `curl` or a `brew`'ed `curl`?

Don't know. `which curl` points to system `curl` but I have `curl` installed via `brew` as well and `brew install --debug` says `[...]Library/Homebrew/shims/shared/curl`

**Comment by miccal** [comment]
What is the 

> > It does for me too -- are you using a system `curl` or a `brew`'ed `curl`?
> 
> Don't know. `which curl` points to system `curl` but I have `curl` installed via `brew` as well and `brew install --debug` says `[...]Library/Homebrew/shims/shared/curl`

What is your output for `which -a curl`? Mine is:
```
|-> which -a curl
/usr/local/opt/curl/bin/curl
/usr/bin/curl
```

**Comment by neutric** [comment]
```
%  which -a curl
/usr/bin/curl
```

**Comment by miccal** [comment]
Thanks @neutric -- and you can confirm this works even with the removal of the `user_agent: :fake`?

**Comment by neutric** [comment]
Correct! Installed fine with below changes. Leaving `binaries/` in `url` works, too, but the link on their homepage is without it.

```
%  git diff
diff --git a/Casks/macspice.rb b/Casks/macspice.rb
index 3cb20023ef..81bd51e16f 100644
--- a/Casks/macspice.rb
+++ b/Casks/macspice.rb
@@ -1,8 +1,8 @@
 cask "macspice" do
-  version "3.1.24,341"
-  sha256 "486baee366773f3589c870d9294e1028c118e29f9bf4dc824322e37f2782475f"
+  version "3.1.25,343"
+  sha256 "22f82c32b6e79916b123a7d1f5ded71a9a30cffc5ab63262b43a0d11db3c45f9"
 
-  url "https://www.macspice.com/mirror/binaries/v#{version.before_comma}/MacSpice3f5.dmg"
+  url "https://www.macspice.com/mirror/v#{version.csv.first}/MacSpice3f5.dmg"
   name "MacSpice"
   homepage "https://www.macspice.com/"
```

**Comment by miccal** [comment]
Now it is not working for me:
```
bash 5.1.12 | /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/casks | 3950
|-> git diff
diff --git a/Casks/macspice.rb b/Casks/macspice.rb
index 3cb20023ef..b1ad6d4496 100644
--- a/Casks/macspice.rb
+++ b/Casks/macspice.rb
@@ -1,9 +1,10 @@
 cask "macspice" do
-  version "3.1.24,341"
-  sha256 "486baee366773f3589c870d9294e1028c118e29f9bf4dc824322e37f2782475f"
+  version "3.1.25,343"
+  sha256 "22f82c32b6e79916b123a7d1f5ded71a9a30cffc5ab63262b43a0d11db3c45f9"
 
-  url "https://www.macspice.com/mirror/binaries/v#{version.before_comma}/MacSpice3f5.dmg"
+  url "https://www.macspice.com/mirror/v#{version.csv.first}/MacSpice3f5.dmg"
   name "MacSpice"
+  desc "Electronic circuit simulator"
   homepage "https://www.macspice.com/"
 
   livecheck do
bash 5.1.12 | /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/casks | 3950
|-> brew fetch macspice
==> Downloading https://www.macspice.com/mirror/v3.1.25/MacSpice3f5.dmg
==> Downloading from https://www.macspice.com/v3.1.25/MacSpice3f5.dmg
curl: (22) The requested URL returned error: 404                              

Error: Download failed on Cask 'macspice' with message: Download failed: https://www.macspice.com/mirror/v3.1.25/MacSpice3f5.dmg
```

**Comment by neutric** [comment]
That would be because I copied from the wrong part in history, sorry. `binaries/` has to stay in `url`, the optional part is `mirror`. So `url "https://www.macspice.com/binaries/v#{version.before_comma}/MacSpice3f5.dmg"`.

**Comment by miccal** [comment]
> That would be because I copied from the wrong part in history, sorry. `binaries/` has to stay in `url`, the optional part is `mirror/`. So `url "https://www.macspice.com/binaries/v#{version.before_comma}/MacSpice3f5.dmg"`.

I saw that and updated my original output -- still not working :(

**Comment by miccal** [comment]
```
|-> /usr/local/opt/curl/bin/curl -sLI https://www.macspice.com/mirror/binaries/v3.1.25/MacSpice3f5.dmg
HTTP/1.1 301 Moved Permanently
Date: Wed, 29 Dec 2021 03:41:02 GMT
Server: Apache
Location: https://www.macspice.com/binaries/v3.1.25/MacSpice3f5.dmg
Cache-Control: max-age=86400
Expires: Thu, 30 Dec 2021 03:41:02 GMT
Vary: Accept-Encoding
Content-Type: text/html; charset=iso-8859-1
X-Cache: MISS from lin-10-170-0-18.gridhost.co.uk
X-Cache-Lookup: HIT from lin-10-170-0-18.gridhost.co.uk:3128
Connection: close
Set-Cookie: DYNSRV=lin-10-170-0-18; path=/

HTTP/1.1 301 Moved Permanently
Date: Tue, 28 Dec 2021 21:32:50 GMT
Server: Apache
Location: https://www.dropbox.com/s/op2w15yfownolim/MacSpice3f5.dmg?dl=1
Cache-Control: max-age=86400
Expires: Wed, 29 Dec 2021 21:32:50 GMT
Vary: Accept-Encoding
Content-Type: text/html; charset=iso-8859-1
Age: 22093
X-Cache: HIT from lin-10-170-0-21.gridhost.co.uk
X-Cache-Lookup: HIT from lin-10-170-0-21.gridhost.co.uk:3128
Connection: close
Set-Cookie: DYNSRV=lin-10-170-0-21; path=/

HTTP/2 301 
cache-control: no-cache,no-cache, no-store
content-security-policy: sandbox
location: /s/dl/op2w15yfownolim/MacSpice3f5.dmg
referrer-policy: strict-origin-when-cross-origin
set-cookie:  gvc=MTI3MjAzMjc1OTAzNTUzMDY3NTEzODUzMjYzODIwNTA4MjM0ODUx; expires=Mon, 28 Dec 2026 03:41:03 GMT; HttpOnly; Path=/; Secure
set-cookie:  t=_yB0qwNMQYJzDOF8Qaaz9nMd; Domain=dropbox.com; expires=Sat, 28 Dec 2024 03:41:03 GMT; HttpOnly; Path=/; Secure
set-cookie:  __Host-js_csrf=_yB0qwNMQYJzDOF8Qaaz9nMd; expires=Sat, 28 Dec 2024 03:41:03 GMT; Path=/; Secure
set-cookie:  locale=en; Domain=dropbox.com; expires=Mon, 28 Dec 2026 03:41:03 GMT; Path=/; Secure
x-content-type-options: nosniff
x-frame-options: DENY
x-permitted-cross-domain-policies: none
x-robots-tag: noindex, nofollow, noimageindex
x-xss-protection: 1; mode=block
content-type: application/json
accept-encoding: identity,gzip
date: Wed, 29 Dec 2021 03:41:03 GMT
server: envoy
strict-transport-security: max-age=31536000; includeSubDomains
strict-transport-security: max-age=31536000; includeSubDomains
strict-transport-security: max-age=31536000; includeSubDomains; preload
vary: Accept-Encoding
x-dropbox-response-origin: far_remote
x-dropbox-request-id: 7eb3f6b364d247fface09ad39995b921

HTTP/2 302 
cache-control: no-cache,no-cache, no-store
content-security-policy: sandbox
location: https://uc0adaf90302f196c097cb9288af.dl.dropboxusercontent.com/cd/0/get/BcvdzilGjBadlqbZBj_ZAUbCdo-8pgd-G6AZObO5x17_FFAq-v3JOrr3G_IZPFyhrU4tEto08Mv-zeBu97zrdK2qd9IBNHbqD5rJM3086PhX245LSzxB7RdxBLBMv6KuyVIuyWB-V75dTcJWjeAvaoF0/file?dl=1#
referrer-policy: strict-origin-when-cross-origin
set-cookie:  locale=en; Domain=dropbox.com; expires=Mon, 28 Dec 2026 03:41:04 GMT; Path=/; secure
set-cookie:  gvc=MzM1NjM1MDE2NjgyNzU4NzExOTc4NTUyODUzMDgzMzg0NDUwNjYz; expires=Mon, 28 Dec 2026 03:41:04 GMT; httponly; Path=/; secure
set-cookie:  t=ACXxoyegjZoab7X5gh0U3CVD; Domain=dropbox.com; expires=Sat, 28 Dec 2024 03:41:04 GMT; httponly; Path=/; secure
set-cookie:  __Host-js_csrf=ACXxoyegjZoab7X5gh0U3CVD; expires=Sat, 28 Dec 2024 03:41:04 GMT; Path=/; secure
x-content-type-options: nosniff
x-frame-options: DENY
x-permitted-cross-domain-policies: none
x-xss-protection: 1; mode=block
content-type: text/html; charset=utf-8
accept-encoding: identity,gzip
date: Wed, 29 Dec 2021 03:41:03 GMT
server: envoy
strict-transport-security: max-age=31536000; includeSubDomains
content-length: 0
strict-transport-security: max-age=31536000; includeSubDomains
strict-transport-security: max-age=31536000; includeSubDomains; preload
x-dropbox-response-origin: far_remote
x-dropbox-request-id: a1ed7938483c4e5b817c9cc26eabba1e

HTTP/2 200 
server: envoy
date: Wed, 29 Dec 2021 03:41:04 GMT
content-type: application/binary
content-length: 3648669
cache-control: max-age=60
pragma: public
x-robots-tag: noindex, nofollow, noimageindex
referrer-policy: no-referrer
etag: 1638749623859136d
x-content-type-options: nosniff
accept-ranges: bytes
content-disposition: attachment; filename="MacSpice3f5.dmg"; filename*=UTF-8''MacSpice3f5.dmg
content-security-policy: sandbox
x-webkit-csp: sandbox
x-content-security-policy: sandbox
vary: Origin
x-server-response-time: 49
strict-transport-security: max-age=31536000; includeSubDomains
strict-transport-security: max-age=31536000; includeSubDomains; preload
x-robots-tag: noindex, nofollow, noimageindex
x-dropbox-response-origin: remote
x-dropbox-request-id: 34f8117da8614f7da9484f5825d41e93
```
```
|-> /usr/bin/curl -sLI https://www.macspice.com/mirror/binaries/v3.1.25/MacSpice3f5.dmg
HTTP/1.1 301 Moved Permanently
Date: Wed, 29 Dec 2021 02:47:51 GMT
Server: Apache
Location: https://www.macspice.com/binaries/v3.1.25/MacSpice3f5.dmg
Cache-Control: max-age=86400
Expires: Thu, 30 Dec 2021 02:47:51 GMT
Vary: Accept-Encoding
Content-Length: 265
Content-Type: text/html; charset=iso-8859-1
Age: 3213
X-Cache: HIT from lin-10-170-0-18.gridhost.co.uk
X-Cache-Lookup: HIT from lin-10-170-0-18.gridhost.co.uk:3128
Connection: close
Set-Cookie: DYNSRV=lin-10-170-0-18; path=/

HTTP/1.1 301 Moved Permanently
Date: Wed, 29 Dec 2021 02:41:48 GMT
Server: Apache
Location: https://www.dropbox.com/s/op2w15yfownolim/MacSpice3f5.dmg?dl=1
Cache-Control: max-age=86400
Expires: Thu, 30 Dec 2021 02:41:48 GMT
Vary: Accept-Encoding
Content-Length: 270
Content-Type: text/html; charset=iso-8859-1
Age: 3578
X-Cache: HIT from lin-10-170-0-18.gridhost.co.uk
X-Cache-Lookup: HIT from lin-10-170-0-18.gridhost.co.uk:3128
Connection: close
Set-Cookie: DYNSRV=lin-10-170-0-18; path=/

HTTP/2 301 
cache-control: no-cache,no-cache, no-store
content-security-policy: sandbox
location: /s/dl/op2w15yfownolim/MacSpice3f5.dmg
referrer-policy: strict-origin-when-cross-origin
set-cookie:  gvc=MTEwNTkxNzUwODQxODg3MjY0MTUxNTk3MTY4NTc4MTY5MDM1MzM4; expires=Mon, 28 Dec 2026 03:41:26 GMT; HttpOnly; Path=/; Secure
set-cookie:  t=9zLLv1veZd66_P5pkMYXNyW5; Domain=dropbox.com; expires=Sat, 28 Dec 2024 03:41:26 GMT; HttpOnly; Path=/; Secure
set-cookie:  __Host-js_csrf=9zLLv1veZd66_P5pkMYXNyW5; expires=Sat, 28 Dec 2024 03:41:26 GMT; Path=/; Secure
set-cookie:  locale=en; Domain=dropbox.com; expires=Mon, 28 Dec 2026 03:41:26 GMT; Path=/; Secure
x-content-type-options: nosniff
x-frame-options: DENY
x-permitted-cross-domain-policies: none
x-robots-tag: noindex, nofollow, noimageindex
x-xss-protection: 1; mode=block
content-type: application/json
accept-encoding: identity,gzip
date: Wed, 29 Dec 2021 03:41:26 GMT
server: envoy
strict-transport-security: max-age=31536000; includeSubDomains
strict-transport-security: max-age=31536000; includeSubDomains
strict-transport-security: max-age=31536000; includeSubDomains; preload
vary: Accept-Encoding
x-dropbox-response-origin: far_remote
x-dropbox-request-id: 252c495e01e74d3e85408c984eb777d1

HTTP/2 302 
cache-control: no-cache,no-cache, no-store
content-security-policy: sandbox
location: https://uc45cd26c9343d40cd0a0870bee0.dl.dropboxusercontent.com/cd/0/get/BcstbvLujNFAweF5hzBAaxCuOSHVnvLql4i2rTQ_rbaTR0ms-N75AEEsTnEPYKR0Rjdivml8BC5Bv3e6z0dSSEWzIKO3kr9rm-8yHPjALLoANNcv_mUWTw9HwT9UXLWRA9oGUFVgisfTf3xQm5yJJPDm/file?dl=1#
referrer-policy: strict-origin-when-cross-origin
set-cookie:  locale=en; Domain=dropbox.com; expires=Mon, 28 Dec 2026 03:41:27 GMT; Path=/; secure
set-cookie:  gvc=MjI0MzM3Nzc0MDY2MTA4OTcxMTU1MzU0MTg4NTY2ODE5Njg3NTg5; expires=Mon, 28 Dec 2026 03:41:27 GMT; httponly; Path=/; secure
set-cookie:  t=0glKQmtfEiVscUq7AsJuyE5l; Domain=dropbox.com; expires=Sat, 28 Dec 2024 03:41:27 GMT; httponly; Path=/; secure
set-cookie:  __Host-js_csrf=0glKQmtfEiVscUq7AsJuyE5l; expires=Sat, 28 Dec 2024 03:41:27 GMT; Path=/; secure
x-content-type-options: nosniff
x-frame-options: DENY
x-permitted-cross-domain-policies: none
x-xss-protection: 1; mode=block
content-type: text/html; charset=utf-8
accept-encoding: identity,gzip
date: Wed, 29 Dec 2021 03:41:27 GMT
server: envoy
strict-transport-security: max-age=31536000; includeSubDomains
content-length: 0
strict-transport-security: max-age=31536000; includeSubDomains
strict-transport-security: max-age=31536000; includeSubDomains; preload
x-dropbox-response-origin: far_remote
x-dropbox-request-id: 95efd32f076e4d7c858189438484fdad

HTTP/2 200 
server: envoy
date: Wed, 29 Dec 2021 03:41:28 GMT
content-type: application/binary
content-length: 3648669
cache-control: max-age=60
pragma: public
x-robots-tag: noindex, nofollow, noimageindex
referrer-policy: no-referrer
etag: 1638749623859136d
x-content-type-options: nosniff
accept-ranges: bytes
content-disposition: attachment; filename="MacSpice3f5.dmg"; filename*=UTF-8''MacSpice3f5.dmg
content-security-policy: sandbox
x-webkit-csp: sandbox
x-content-security-policy: sandbox
vary: Origin
x-server-response-time: 35
strict-transport-security: max-age=31536000; includeSubDomains
strict-transport-security: max-age=31536000; includeSubDomains; preload
x-robots-tag: noindex, nofollow, noimageindex
x-dropbox-response-origin: remote
x-dropbox-request-id: a996f8f824c743458d2588906b19cb5f
```

**Comment by miccal** [comment]
I am confused and tired -- I will have another look at this later.

**Comment by miccal** [comment]
Ok, now it is working for me locally.

So, we have confirmed this works as intended with the system `curl` and a `brew`'ed `curl`.

Most likely this is an intermittent and region-specific failure with the Dropbox redirects.

So I am going to merge this and we will see how it goes.

Thanks @alxrey.

**Comment by alxrey** [comment]
Thanks to you. But, for information, it still doesn't work for me with `brew install macspice` (403 error). However it works with `brew install macspice --debug`.

Unfortunately I don't have enough knowledge of the Homebrew system to investigate further.

**Comment by neutric** [comment]
@alxrey That sounds to me like the intermittent and region-specific failures with Dropbox redirects that @miccal suspected.

**Pull request merged**